common/app/com/commercetools/sunrise/sessions/wishlist/DefaultWishlistInSession.java

```diff
@@ -33,17 +41,24 @@ public DefaultWishlistInSession(final Configuration configuration, final ObjectS
     }
 
     @Override
+    public Optional<WishlistViewModel> findWishlist() {
+        return session.findObjectByKey(wishlistSessionKey, WishlistViewModel.class);
+    }
+
+    @Override
     protected Logger getLogger() {
         return LOGGER;
     }
 
     @Override
-    protected void storeAssociatedData(@NotNull final ShoppingList value) {
-        session.overwriteValueByKey(wishlistIdSessionKey, value.getId());
+    protected void storeAssociatedData(@NotNull final ShoppingList wishlist) {
+        session.overwriteValueByKey(wishlistIdSessionKey, wishlist.getId());
+        session.overwriteObjectByKey(wishlistSessionKey, wishlistViewModelFactory.create(wishlist));
```

Comment:
Having the mini-wishlist and mini-cart in the session together won't work anymore, even with the product limit. It's been some time since I wanted to use cache by default, because everybody ends up with the same space issue. If we merge this PR we will need to address the session issue before releasing...